### DISCUSSION TITLE
->save() - "Unable to create a directory at"

### DISCUSSION THREAD

**Discussion by Teglgaard** [spatie/image#175] [Q&A]
Hi I am in need of support figuring out if there is a bug in the library.

I am trying to run the following line as part of a job which fails with below error message stating it is "Unable to create a directory at public" . 

`Image::load("public/".$image)->fit(Manipulations::FIT_MAX,250,250)->save("public/".$thumb_base_file_name);
`
The thumb_base_file_name comes from
$thumb_base_file_name = Storage::disk('s3')->put('image', $request['image']);

I am running Laravel 9 on AWS Elastic Beanstalk PHP 8.0.24

My storage and public folder is set with chmod 777.
CRON is being run as user "webapp" and the  whole root folder with contents is all owned by user webapp.

Does any one have any idea of what might be the cause? Any support is appreciated.

```
League\Flysystem\UnableToCreateDirectory: Unable to create a directory at public. in /var/app/current/vendor/league/flysystem/src/UnableToCreateDirectory.php:20
Stack trace:
#0 /var/app/current/vendor/league/flysystem/src/Local/LocalFilesystemAdapter.php(330): League\Flysystem\UnableToCreateDirectory::atLocation('public', '')
#1 /var/app/current/vendor/league/flysystem/src/Local/LocalFilesystemAdapter.php(122): League\Flysystem\Local\LocalFilesystemAdapter->ensureDirectoryExists('public', 448)
#2 /var/app/current/vendor/league/flysystem/src/Local/LocalFilesystemAdapter.php(112): League\Flysystem\Local\LocalFilesystemAdapter->ensureRootDirectoryExists()
#3 /var/app/current/vendor/league/glide/src/ServerFactory.php(89): League\Flysystem\Local\LocalFilesystemAdapter->__construct('public')
#4 /var/app/current/vendor/league/glide/src/ServerFactory.php(55): League\Glide\ServerFactory->getSource()
#5 /var/app/current/vendor/league/glide/src/ServerFactory.php(342): League\Glide\ServerFactory->getServer()
#6 /var/app/current/vendor/spatie/image/src/GlideConversion.php(114): League\Glide\ServerFactory::create(Array)
#7 /var/app/current/vendor/spatie/image/src/GlideConversion.php(68): Spatie\Image\GlideConversion->createGlideServer('public/16660897...', NULL)
#8 /var/app/current/vendor/spatie/image/src/Image.php(120): Spatie\Image\GlideConversion->performManipulations(Object(Spatie\Image\Manipulations))
#9 /var/app/current/app/Jobs/ProcessResizeImages.php(49): Spatie\Image\Image->save('storage\\app\\pub...')

```

**Comment by DoubleClickDesignLtd** [comment]
Did you manage to get this sorted? I'm having a similiar issue

**Comment by Teglgaard** [comment]
Thanks for asking. 

Yes I did it was an ownership of the directory problem. So you need to ensure correct permissions and ownership of the directory you are saving to.
